vpr/src/place/place_util.h

```diff
@@ -218,4 +218,7 @@ void load_grid_blocks_from_block_locs();
 
 ///@brief Builds legal_pos structure. legal_pos[type->index] is an array that gives every legal value of (x,y,z) that can accommodate a block.
 void alloc_and_load_legal_placement_locations(std::vector<std::vector<std::vector<t_pl_loc>>>& legal_pos);
+
+///@brief Marks the location and grid usage for a block that is set at a location
```

Comment:
Should also mention that it performs error checking on whether this is a legal location or not for a block of that type, and prints and error if it is not legal.